app/models/king.rb

```diff
@@ -21,4 +21,8 @@ def valid_move?
 
   end
 
+  def piece_exists?(x, y)
+    Piece.where(game_id: self.id, row_position: x, col_position: y).exists?
+  end
+
```

Comment:
This method would move to piece.rb so that each of the subclasses could call it.  However, as we mentioned in our meeting it probably belongs in game.rb instead.